File: app/src/latest/java/tgx/flavor/NLoader.java
package tgx.flavor;

import org.thunderdog.challegram.BuildConfig;
import org.thunderdog.challegram.N;

public class NLoader {
  private static boolean loaded;

  public static String collectLog () {
    return "";
  }

  public static synchronized boolean loadLibraries () {
    if (loaded) {
      return true;
    }
    if (BuildConfig.SHARED_STL) {
      System.loadLibrary("c++_shared");
    }
    System.loadLibrary("cryptox");
    System.loadLibrary("sslx");
    System.loadLibrary("tdjni");
    System.loadLibrary("leveldbjni");
    System.loadLibrary("tgcallsjni");
    System.loadLibrary("tgxjni");
    N.setupLibraries();
    loaded = true;
    return true;
  }
}

File: app/src/sinceMarshmallow/kotlin/tgx/flavor/NLoader.kt
@file:JvmName("NLoader")

package tgx.flavor

import androidx.tracing.trace
import org.thunderdog.challegram.BuildConfig
import org.thunderdog.challegram.N


fun collectLog(): String = ""

private fun loadLibrary(name: String) = trace("load:$name") {
  System.loadLibrary(name)
}

@Synchronized
fun loadLibraries() {
  if (BuildConfig.SHARED_STL) {
    loadLibrary("c++_shared")
  }
  loadLibrary("cryptox")
  loadLibrary("sslx")
  loadLibrary("tdjni")
  loadLibrary("leveldbjni")
  loadLibrary("tgcallsjni")
  loadLibrary("tgxjni")
  trace("setup") {
    N.setupLibraries()
  }
}
